components/tor/tor_launcher_factory.cc

```diff
@@ -130,7 +133,7 @@ void TorLauncherFactory::OnTorControlCheckComplete() {
 }
 
 void TorLauncherFactory::KillTorProcess() {
-  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  SEQUENCE_CHECKER(sequence_checker_);
   if (tor_launcher_.is_bound())
     tor_launcher_->Shutdown();
   control_->Stop();
```

Comment:
Basically, it is illegal to call `Stop()` on this sequence, especially given that this could happen at any arbitrary moment (i.e. closing a last tor window). This way `Stop()` modifies `running_` and `async_events_` on this sequence, while they are also accessed/modified on another sequence. So I gather we should post this to IO as well, and in turn `Stop()` can call `Error()` synchronously

Reply:
fixed in aeed04b506e574973c5df12c9781dde63e187342